client_manager.go

```diff
@@ -8,6 +8,8 @@ import (
 	"time"
 )
 
+var mu sync.Mutex
```

Comment:
why so? what if would i want to create a several ClientManager? Mutex must be in structure. Also right way to determine mutex - paste above variables, that should be protected

https://golang.org/doc/effective_go.html
>Grouping can also indicate relationships between items, such as the fact that a set of variables is protected by a mutex.

```go
        ...

        mu    sync.Mutex
 	cache map[[sha1.Size]byte]*list.Element
        
        ...
```